### DISCUSSION TITLE
`reloadOnChange: true`: How to configure for multiple ocelot files?

### DISCUSSION THREAD

**Discussion by neetra** [ThreeMammals/Ocelot#2020] [Q&A] [answered]
For using [multiple configuration files](https://ocelot.readthedocs.io/en/latest/features/configuration.html#merging-configuration-files) , we use addOcelot, how to allow [ Reload JSON config on change](https://ocelot.readthedocs.io/en/latest/features/configuration.html#reload-json-config-on-change)

**Comment by AbbasKhalili** [comment]
I have the same problem

**Comment by raman-m** [comment]
@neetra 
The third boolean parameter in code sample is exactly the flag:
`.AddJsonFile($"appsettings.{hostingContext.HostingEnvironment.EnvironmentName}.json", true, true)`

See definition:
```csharp
public static IConfigurationBuilder AddJsonFile(this IConfigurationBuilder builder,
    string path,
    bool optional,
    bool reloadOnChange) // <-- this one!
{ }
```

**Comment by raman-m** [comment, marked as answer]
Hi Netra!
I have finally got your question: **How to configure for multiple ocelot files?**

In the current [February'24](https://github.com/ThreeMammals/Ocelot/milestone/5) milestone we worked on the #1216 issue and PR #1227.
So, `ConfigurationBuilderExtensions` were updated and appropriate merging logic.
For example in [this AddOcelot](https://github.com/ThreeMammals/Ocelot/pull/1227/files#diff-b940a9c08ab13a6c40e0141da2de2f03ae84fe3d9d86586c470ba3eddbbbf649R72-R73) overloaded method the new optional parameters are introduced to have control over `reloadOnChange` argument for [this final injection](https://github.com/ThreeMammals/Ocelot/pull/1227/files#diff-b940a9c08ab13a6c40e0141da2de2f03ae84fe3d9d86586c470ba3eddbbbf649R200).

Current [Configuration](https://github.com/ThreeMammals/Ocelot/blob/develop/docs/features/configuration.rst) docs are available in develop branch.
The updated [Merging Configuration Files](https://github.com/ThreeMammals/Ocelot/blob/develop/docs/features/configuration.rst#merging-configuration-files1) and [Reload JSON Config On Change](https://github.com/ThreeMammals/Ocelot/blob/develop/docs/features/configuration.rst#reload-json-config-on-change) will be both delivered by PR #1227 as a part of [Version 23.2](https://github.com/ThreeMammals/Ocelot/milestone/5) soon...

Hope it helps!

### P.S. 
You need to understand that after Merging stage into one primary `ocelot.json` file only this file is under tracking by the configuration provider. The next merging will perform when new Ocelot restart.
If you want to upload configuration parts JSON-files, you can upload using [Administration](https://ocelot.readthedocs.io/en/latest/features/administration.html) feature, but you have to restart Ocelot instance to merge all available part and env files into one primary file.
